08_character_input_output/exercises/code.c

```diff
@@ -0,0 +1,34 @@
+// Write a program that reads input as a stream of characters until encountering EOF. Have
+// the program print each input character and its ASCII decimal value. Note that characters
+// preceding the space chatacter in the ASCII sequence are nonprinting characters.
+// Treat them specially. If the nonprinting character is a newline or tab, print \n or \t,
+// respectively. Otherwise, use control-character notation. For instance, ANSII 1 is Ctrl+A,
+// which can be displayed as ^A. Note that the ASCII value for A is value for Ctrl+A
+// plus 64. A similar relation holds for the other nonprinting characters. Print 10 pairs per
+// line, except start a fresh line each time a newline character is encountered. (Note: The
+// operating system may have special interpretations for some control characters and keep
+// them from reaching the program.)
+#include <stdio.h>
+
+int main(void)
+{
+    char ch;  
+    int count = 0; 
+    while ((ch = getchar()) != EOF)
+    {
+	count++;
+        if (ch =='\n')
+	    printf("\\n = %d.\n", ch);
+        else if (ch == '\t')
+            printf("\\t = %d.\n", ch);
+	else if (ch < 32)
+	    printf("^%c = %d\n", ch + 64, ch); 
+	else if (count == 11)
```

Comment:
This will work only one time. E.g. if you have 20 characters it will produce 2 lines, which is correct. But if you enter 30 characters, it will also produce 2 line, although 3 is expected.